PIMbench/cpp-vgg13/README.md

```diff
@@ -0,0 +1,97 @@
+# VGG13 
+
+VGG13 is a deep convolutional neural network architecture with 10 convolutional layers and 3 fully connected layers designed for image classification and feature extraction with a focus on simplicity and depth. Each convolutional layer and fully connected layer (except the last one) is followed by an ReLU activation function to introduce non-linearity. There is also a max pool layer after every two convolutional layers for a total of 5 max pooling layers.
+
+## Directory Structure
+```
+cpp-vgg13/
+├── PIM/
+│   ├── Makefile
+│   ├── vgg13.cpp
+│   ├── vgg13_weights.py
+├── baselines/
+│   ├── vgg13.py
+├── README.md
+├── Makefile
+├── slurm.h
+```
+
+## Implementation Description
+
+This repository contains two different implementations of the VGG13 benchmark:
+1. CPU & GPU
+2. PIM
+
+### Baseline Implementation
+
+CPU and GPU have been used as baselines.
+
+#### CPU & GPU
+
+The CPU and GPU variants of VGG13 have been implemented using standard Python and PyTorch framework for machine learning. The same script is used for both, the device is specified from the command line by the user. The script performs inference using a pre-trained VGG13 model on the CIFAR-100 dataset, supporting both CPU and CUDA-enabled GPU devices. It preprocesses the data, adjusts the model's classifier for 100 classes, and optimizes it with TorchScript. The script evaluates the model's top-1 and top-5 accuracy and measures the execution time per image. Results, including accuracy percentages and timing, are printed at the end. Command-line arguments specify the device and the number of test images which is by default 1000.
+
+### PIM Implementation
+
+The PIM variant is implemented using C++ and three different PIM architectures - BITSIMD, Fulcrum and Bank-level can be tested with this. 
+  
+## Compilation and Execution Instructions for Specific Variants
+
+### CPU Variant
+
+To run the script for the CPU variant, use command like the following example:
+
+```bash
+cd baselines
+python3 vgg13.py -s cpu -n 100
```

Comment:
Could you please elaborate what does `n` flag mean?

Reply:
Updated the README file. Kindly check.